flask/user.py

```diff
@@ -0,0 +1,339 @@
+            "createdAt": user["createdAt"],
+            "packages": response_packages,
+        }
+        return (
+            jsonify(
+                {
+                    "message": "User found",
+                    "user": user_account,
+                    "packages": response_packages,
+                    "code": 200,
+                }
+            ),
+            200,
+        )
+    else:
+        return jsonify({"message": "User not found", "code": 404}), 404
+
+
+@app.route("/users/delete", methods=["POST"])
+@swag_from("documentation/delete_user.yaml", methods=["POST"])
+def delete_user():
+    uuid = request.form.get("uuid")
+    password = request.form.get("password")
+    username = request.form.get("username")
+
+    if not uuid:
+        return jsonify({"message": "Unauthorized", "code": 401}), 401
+    else:
+        user = db.users.find_one({"uuid": uuid})
+
+    if not user:
+        return "Invalid email or password", 401
+
+    if password:
+        password += salt
+        hashed_password = hashlib.sha256(password.encode()).hexdigest()
+        if hashed_password != user["password"]:
+            return jsonify({"message": "Invalid email or password", "code": 401}), 401
+        else:
+            db.users.delete_one({"uuid": uuid})
+            return jsonify({"message": "User deleted", "code": 200}), 200
+
+    elif username and "admin" in user["roles"]:
+        delete_user = db.users.find_one({"username": username})
+        if delete_user:
+            db.users.delete_one({"username": username})
+            return jsonify({"message": "User deleted", "code": 200}), 200
+        else:
+            return jsonify({"message": "User not found", "code": 404}), 404
+
+    else:
+        return jsonify({"message": "Unauthorized", "code": 401}), 401
+
+
+@app.route("/users/account", methods=["POST"])
+@swag_from("documentation/account.yaml", methods=["POST"])
+def account():
+    uuid = request.form.get("uuid")
+    if not uuid:
+        return jsonify({"message": "Unauthorized", "code": 401}), 401
+    else:
+        user = db.users.find_one({"uuid": uuid})
+
+    if not user:
+        return jsonify({"message": "User not found", "code": 404}), 404
+
+    user_account = {
+        "username": user["username"],
+        "email": user["email"],
+        "createdAt": user["createdAt"],
+        "loginAt": user["loginAt"],
+        "lastLogout": user["lastLogout"],
+    }
+    return jsonify({"message": "User Found", "user": user_account, "code": 200}), 200
+
+
+@app.route("/users/admin", methods=["POST"])
+@swag_from("documentation/admin.yaml", methods=["POST"])
+def admin():
+    uuid = request.form.get("uuid")
+    if not uuid:
+        return jsonify({"message": "Unauthorized", "code": 401}), 401
+    else:
+        user = db.users.find_one({"uuid": uuid})
+
+    if not user:
+        return jsonify({"message": "User not found", "code": 404}), 404
+    
+    if "admin" not in user["roles"]:
```

Comment:
How are we differentiating if this a sudo admin or namespace admin ?

Reply:
only the users who are `sudoadmin` would only have the `admin` in roles, in the namespaces we store the `namespace admins` in namespace document ( not in the user document).